test/integration/builder/inserts.js

```diff
@@ -1302,5 +1302,17 @@ module.exports = function(knex) {
           });
       });
     });
+
+    describe('#3553 insert with bigint', function() {
+      if (typeof BigInt === 'undefined') return;
+
+      it('should allow insert with BigInt', async function() {
+        await knex('accounts').insert([{ id: BigInt(12) }]);
```

Comment:
Might want to:

1. Test w/ a number that is really large.  (Ex:  `BigInt("1234567812345678123456781234567812345678")`)
2. Read the number back out of the database as well.

Also: it looks like the `MSSQL` changes the datatype depending upon the size of the integer.  So, there are at least 5 different scenarios that should be tested there.  (And, probably more like 7-9 cases if you count the boundary conditions as well).